src/Microsoft.AspNet.OData.Shared/Builder/ParameterConfiguration.cs

```diff
@@ -51,5 +51,44 @@ protected ParameterConfiguration(string name, IEdmTypeConfiguration parameterTyp
         /// Gets or sets a value indicating whether this parameter is nullable or not.
         /// </summary>
         public bool Nullable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this parameter is optional or not.
+        /// </summary>
+        public bool Optional { get; set; }
+
+        /// <summary>
+        /// Gets or sets a default value for optional parameter.
+        /// </summary>
+        public string DefaultValue { get; set; }
+
+        /// <summary>
+        /// Sets the optional value as true.
+        /// </summary>
+        public ParameterConfiguration IsOptional()
+        {
+            Optional = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the optional value as false.
+        /// </summary>
+        public ParameterConfiguration IsRequired()
+        {
+            Optional = false;
+            return this;
+        }
+
+        /// <summary>
+        ///  Sets the optional value as true, default value as given value.
+        /// </summary>
+        /// <param name="defaultValue">The default value.</param>
+        public ParameterConfiguration HasDefaultValue(string defaultValue)
```

Comment:
I think SetDefaultValue might be more clear. Please feel free to ask the team for their opinion as well.

Reply:
Also, should we replace the DefaultValue set method with this function? Otherwise, should the DefaultValue set method be private?

Reply:
That's the Web API OData naming convention, I think. for example: there are a lot of fluent api names:

1. HasActionLink
2. HasFunctionLink
3. HasManyPath
4. HasSingleton
….

Reply:
Got it, thanks for the clarification